includes/class-wcpdf-main.php

```diff
@@ -304,11 +308,13 @@ public function generate_pdf_ajax() {
 		}
 
 		if ( empty( $_REQUEST['order_ids'] ) ) {
-			wp_die( esc_attr__( "You haven't selected any orders", 'woocommerce-pdf-invoices-packing-slips' ) );
+			$message = esc_attr__( "You haven't selected any orders", 'woocommerce-pdf-invoices-packing-slips' );
+			wcpdf_safe_redirect_or_die( $redirect_url, $message );
 		}
 
 		if ( empty( $_REQUEST['document_type'] ) ) {
-			wp_die( esc_attr__( 'Some of the export parameters are missing.', 'woocommerce-pdf-invoices-packing-slips' ) );
+			$message = esc_attr__( 'Some of the export parameters are missing.', 'woocommerce-pdf-invoices-packing-slips' );
+			wcpdf_safe_redirect_or_die( $redirect_url, $message );
```

Comment:
Same here.
And there are a few below that I don't mention to prevent repetition.